spec/lib/mercury/mercury_spec.rb

```diff
@@ -75,12 +75,85 @@
     end
   end
 
-  def with_mercury(&block)
+  # Commented out until this gets fixed: https://github.com/eventmachine/eventmachine/issues/670
+  # it 'raises an error when a connection cannot be established' do
+  #   expect { em { Mercury.open(port: 31999) { done } } }.to raise_error /Failed to establish connection/
+  #   expect(EM.reactor_running?).to be false
+  # end
+
+  it 'raises an error when the connection breaks' do
+    expect { em { Mercury.open { done } } }.to raise_error /Lost connection/
+    expect(EM.reactor_running?).to be false   # make sure we're not triggering EventMachine cleanup bugs
+  end
+
+  it 'does not obscure exceptions thrown inside the reactor' do
+    expect { em { Mercury.open { raise 'oops' } } }.to raise_error 'oops'
+    expect(EM.reactor_running?).to be false   # make sure we're not triggering EventMachine cleanup bugs
+  end
+
+  describe '#publish' do
+    context 'docker assumptions' do
```

Comment:
Ideally, yes. Since mercury is officially decoupled from ascent, I'd still like to keep this "test" here. But we could have similar integration tests, probably at the service level. "Assert that if the RabbitMQ server goes down during window W in housekeeper, then housekeeper crashes but the request it was processing eventually completes." The problem is that in integration tests it's not natural to have such finely-grained control of services that you can shut one down within a precise window of execution in another one.